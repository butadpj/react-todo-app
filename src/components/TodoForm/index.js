import React from 'react';

const TodoForm = (props) => {
  return (
    <form className="todo-form" onSubmit={props.onSubmit}>
      <input 
        className="todo__input" 
        type="text" 
        value={props.value} 
        placeholder="Add your new todo" 
        onChange={props.onChange}
      />
      <button type="submit" className="add-btn btn">Add</button>
    </form>
  )
}

export default TodoForm