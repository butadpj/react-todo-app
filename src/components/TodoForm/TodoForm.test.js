import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import TodoForm from './index';

describe("<TodoForm />", () => {
  it("renders without crashing", () => {
    shallow(<TodoForm />);
  });

  it("UI doesn't change unexpectedly", () => {
    const wrapper = shallow(<TodoForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('input value is based on props', () => {
    const onChangeMock = jest.fn()
      .mockImplementation((input, component) => 
        component.setProps({ value: input }));

    const wrapper = mount(
      <TodoForm 
        value={'New Todo'} 
        onChange={() => onChangeMock('Newest Todo', wrapper)}
      />
    );

    // Input's value should be -> 'New Todo'
    expect(wrapper.find('input').props().value).toBe('New Todo');

    // Call onChangeMock() -- Change props.value from 'New Todo' -> 'Newest Todo'
    wrapper.find('input').simulate('change');

    // Input's value should be -> 'Newest Todo'
    expect(wrapper.find('input').props().value).toBe('Newest Todo');
  });

  it('clears the input after submitting form', () => {
    const onChangeMock = jest.fn();
    
    const onSubmitMock = jest.fn()
      .mockImplementation(component => 
        component.setProps({ value: '' })); 

    const wrapper = mount(
      <TodoForm 
        value={'New Todo'} 
        onChange={onChangeMock}
        onSubmit={() => onSubmitMock(wrapper)}
      />
    );

    // Input's value should be -> 'New Todo'
    expect(wrapper.find('input').props().value).toBe('New Todo');

    // Call onChangeMock() -- Change props.value from 'New Todo' -> 'Newest Todo'
    wrapper.find('form').simulate('submit');

    // Input's value should be -> 'Newest Todo'
    expect(wrapper.find('input').props().value).toBe('');
  });
});
