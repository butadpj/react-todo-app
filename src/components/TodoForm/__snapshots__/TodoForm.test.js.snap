// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoForm /> UI doesn't change unexpectedly 1`] = `
<form
  className="todo-form"
>
  <input
    autoFocus={true}
    className="todo__input"
    placeholder="Add your new todo"
    type="text"
  />
  <button
    className="btn btn--success add-btn"
    type="submit"
  >
    Add
  </button>
</form>
`;
